### PR TITLE
Drop redundant image fields

### PR DESCRIPTION
**Pull request by kargakis**
/cc jim-minter pweil-

### CODE REVIEW

File: pkg/api/agentPoolOnlyApi/vlabs/validate.go [hunk 1]
```diff
@@ -15,21 +14,8 @@ func init() {
 	validate = validator.New()
 }
 
-func validateImageNameAndGroup(name, resourceGroup string) error {
```
Comment (pweil-) [inline pkg/api/agentPoolOnlyApi/vlabs/validate.go]: should this have moved upstream into validation?  Seems like we still want it

Comment (kargakis) [reply pkg/api/agentPoolOnlyApi/vlabs/validate.go]: We dropped anything related to the agentPoolApi - see https://github.com/Azure/acs-engine/pull/2513

Comment (kargakis) [reply pkg/api/agentPoolOnlyApi/vlabs/validate.go]: https://github.com/Azure/acs-engine/pull/2513#issuecomment-377609537

Comment (pweil-) [reply pkg/api/agentPoolOnlyApi/vlabs/validate.go]: I probably should've put the comment on the other api object.  I missed the other validation piece but see it in the upstream PR.  Disregard

**Comment by kargakis** [comment]
> don't forget the TestValidateImageNameAndGroup

We have this now from upstream; the agentPoolApi version of the test was never added and I hadn't added any tests in my fork.

**Comment by pweil-** [comment]
> don't forget the TestValidateImageNameAndGroup

and since agentPoolApi doesn't have the test this is

/lgtm

**Pull request merged**